modules/IsabellenhuetteIemDcr/main/powermeterImpl.cpp

```diff
@@ -0,0 +1,281 @@
+}
+
+types::powermeter::TransactionStopResponse powermeterImpl::handle_stop_transaction(std::string& transaction_id) {
+    // your code for cmd stop_transaction goes here
+    types::powermeter::TransactionStopResponse return_value;
+
+    stop_transaction_running.store(true);
+    EVLOG_info << "handle_stop_transaction() called.";
+
+    if(is_initialized.load() == false) {
+        return_value.status = types::powermeter::TransactionRequestStatus::UNEXPECTED_ERROR;
+        return_value.error = "Init of communication not finished yet.";
+        EVLOG_error << "Aborted: " << *return_value.error;
+    } else if (transaction_active.load() == true) {
+        try {
+            // Stop transaction
+            this->controller->call_with_retry([this]() { this->controller->post_receipt("E"); },
+                                              mod->config.resilience_transaction_request_retries,
+                                              mod->config.resilience_transaction_request_retry_delay);
+            // Wait for signature calculation
+            std::this_thread::sleep_for(std::chrono::milliseconds(250));
+            // Read receipt
+            return_value.signed_meter_value =
+                this->controller->call_with_retry([this]() { return this->controller->get_receipt(); },
+                                                  mod->config.resilience_transaction_request_retries,
+                                                  mod->config.resilience_transaction_request_retry_delay);
+            // Prepare positive response
+            return_value.status = types::powermeter::TransactionRequestStatus::OK;
+            return_value.error = "";
+        } catch (const std::exception& e) {
+            return_value.status = types::powermeter::TransactionRequestStatus::UNEXPECTED_ERROR;
+            return_value.error = e.what();
+            EVLOG_error << "Aborted: " << return_value.error.value_or("");
+        }
+    } else {
+        // No transaction running. So return last transaction (if available)
+        try {
+            return_value.signed_meter_value = this->controller->get_transaction();
+            return_value.status = types::powermeter::TransactionRequestStatus::OK;
+            return_value.error = "";
+        } catch (std::exception& e) {
+            return_value.status = types::powermeter::TransactionRequestStatus::UNEXPECTED_ERROR;
+            return_value.error = std::string(e.what()) + " Maybe no transaction to return?";
+            EVLOG_warning << "Aborted: " << return_value.error.value_or("");
+        }
+    }
+
+    stop_transaction_running.store(false);
+
+    return return_value;
+}
+
+void powermeterImpl::check_config() {
+    if (mod->config.ip_address.length() <= 0) {
+        EVLOG_error << "Incorrect module config: parameter ip_address is empty." << std::endl;
+        throw std::runtime_error("ip_address invalid. Please check configuration.");
+    }
+    if (mod->config.port_http < 0) {
```

Comment:
You can here use the comfort of the manifests to specify minimum and maximum values. Then you don't need to check this stuff here.